src/server/acceptor.rs

```diff
@@ -35,16 +34,16 @@ impl<A: NetworkAcceptor> AcceptorPool<A> {
         for _ in 0..threads { spawn() }
 
         // Spawn the supervisor
-        Thread::scoped(move || for () in supervisor_rx.iter() { spawn() })
+        thread::scoped(move || for () in supervisor_rx.iter() { spawn() })
```

Comment:
To make this blocking we'd just remove the `thread::scoped` call here and just run this loop.